test/mock_query_test.dart

```diff
@@ -312,6 +312,35 @@ void main() {
     ));
   });
 
+  test('orderBy descending returns documents with null fields first', () async {
```

Comment:
I have a question on the real Firestore behavior though. Does `descending: true` and `descending: false` both return null values first? Cause the way we implemented `orderBy` in fake_cloud_firestore, it returns null values first no matter the value of `descending`. See https://github.com/atn832/fake_cloud_firestore/blob/01e77ce065d6648a5dd8ba31b8296dd34217a5d2/lib/src/mock_query.dart#L93-L102

Reply:
Let me check 🤔

Reply:
Looks like the doc with null value on the order field will skip and will not list out😅

Reply:
FYR: https://firebase.google.com/docs/firestore/query-data/order-limit-data

Note: An orderBy() clause also filters for existence of the given field. The result set will not include documents that do not contain the given field.

Reply:
WOW. Good find

Reply:
I also got this:

https://firebase.google.com/docs/firestore/manage-data/data-types

Value type ordering

When a query involves a field with values of mixed types, Cloud Firestore uses a deterministic ordering based on the internal representations. The following list shows the order:

Null values
Boolean values
Integer and floating-point values, sorted in numerical order
Date values
Text string values
Byte values
Cloud Firestore references
Geographical point values
Array values
Map values

Reply:
In order to keep this PR simple and to quickly and safely fix #180, maybe we could make a few separate PRs:
- in this PR, you can fix the exception when a null value is converted to a Comparable. You can then simplify your unit tests to not include docs with missing attributes. Perhaps you could just reuse the existing test at https://github.com/atn832/fake_cloud_firestore/blob/01e77ce065d6648a5dd8ba31b8296dd34217a5d2/test/mock_query_test.dart#L297 and put the doc with null value in first position.
- in a separate PR, we can filter out docs with missing attributes.
- in the far future, we could separately handle whether the order for mixed types is inverted when using descending or not (if `descending: true`, do null values appear first or last?), and handle mixed types more exhaustively.